src/admin/UserForm.js

```diff
@@ -0,0 +1,114 @@
+import {
+  Grid, TextField, Button, InputLabel,
+} from '@material-ui/core';
+import Select from 'react-select';
+import React from 'react';
+import { useQuery, useMutation } from '@apollo/client';
+import useStyles from './useStyles';
+import { GET_USER_ROLES, CREATE_USER, UPDATE_USER } from '../queries/UserQueries';
+import Info from '../Info';
+
+export const validateUser = (
+  username, firstName, lastName, password, userRole, callback,
+) => (
+  callback(!!(username && firstName && lastName && password && userRole)));
+
+function UserForm({ user, callApi, errorAutoHide = 2500 }) {
+  const [username, setUsername] = React.useState(user?.username || '');
```

Comment:
The question is why would you need to split your user object into separate states when you can have one `const [newUser, setNewUser] = React.useState(user)`. You can then set specific props of the user in your input event handles ie `onChange={ (e) => setNewUser({ ...newUser, firstName: e.target.value }) }`